### PR TITLE
Add upper bound to prevent usage of NumPy 2

### PR DESCRIPTION
**Pull request by bdice**
NumPy 2 is expected to be released in the near future. For the RAPIDS 24.04 release, we will pin to `numpy>=1.23,<2.0a0`. This PR adds an upper bound to affected RAPIDS repositories.

xref: https://github.com/rapidsai/build-planning/issues/29

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Bradley! 🙏

Could we add this to wheels too? Looks like it is Conda only atm

**Comment by bdice** [comment]
Wheels added in d7d067e.

**Comment by jakirkham** [comment]
Looks good. Thanks Bradley! 🙏

**Review by raydouglass — APPROVED** [review]

File: python/cucim/pyproject.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ dependencies = [
     "cupy-cuda11x>=12.0.0",
     "lazy_loader>=0.1",
     "numpy",
+    "numpy>=1.23.4,<2.0a0",
```
Comment (raydouglass) [inline python/cucim/pyproject.toml:30]: Probably doesn't matter, but something is causing numpy to appear twice.

**Review by grlee77 — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**